impeller/compiler/compiler.cc

```diff
@@ -35,6 +35,42 @@ static CompilerBackend CreateMSLCompiler(const spirv_cross::ParsedIR& ir,
   sl_options.msl_version =
       spirv_cross::CompilerMSL::Options::make_msl_version(1, 2);
   sl_compiler->set_msl_options(sl_options);
+
+  // Set metal resource mappings to be consistent with location based mapping
+  // used on other backends when creating fragment shaders.
+  if (source_options.remap_samplers) {
```

Comment:
Can this just be the default?

Reply:
There are still some compilation errors when using this on the flutter/engine shaders. I need to spend some more time tracking down why that is

Reply:
SGTM. I'm just catching up on the issue and the conversation in chat. Will take a closer look in the morning.

Reply:
I think this works if I also set `enable_decoration_binding` to true. that also fixes the issue with floats being out of order, but I need to verify that our bindings are still correct.